Adapt Istio roles (#1749)
kyma-project/istio: docs/user/technical-reference/05-10-istio-controller-rbac.md

As Istio Controller orchestrates the deployment of Istio components, it necessitates comprehensive management privileges for Istio resources. 
These privileges must mirror the access control levels accorded to the resources themselves, ensuring seamless operation.

> ## Usage of wildcards in ClusterRole definitions

Reviewer: ```suggestion
## Usage of Wildcards in ClusterRole Definitions
```

### ClusterRole istio-manager-role

> This ClusterRole allows full management of all Istio resources. This is done with the usage of the wildcard "*" to specify all resources within the Istio API group.

Reviewer: ```suggestion
This ClusterRole allows full management of all Istio resources. This is achieved by using the wildcard "*" to specify all resources within the Istio API group.
```

This is relevant for the Istio Controller to be able to create, update, and delete Istio resources as needed, as well as pass the necessary permissions to Istio components.

> Additionally, this role allows read access to all resources from:

Reviewer: ```suggestion
Additionally, this role allows read access to all resources from the following groups:
```

- gateway.networking.k8s.io
- networking.k8s.io
- rbac.istio.io

This is required for Istio installation to succeed.
Lastly, this role allows full management over ingresses/status resources from networking.k8s.io API group.

> This is required by Istio installation to pass the necessary permissions to Istiod to manage the status of ingress resources.

Reviewer: ```suggestion
This is required for Istio installation to succeed.
Lastly, this role allows full management over ingresses and status resources from the `networking.k8s.io` API group.
Istio installation requires the role to pass the necessary permissions to Istiod, which manages the status of ingress resources.
```

> ### ClusterRole istio-reader-clusterrole-istio-system

Reviewer: ```suggestion
### ClusterRole `istio-reader-clusterrole-istio-system`
```

This ClusterRole grants read-only access to all Istio resources. This is done with the usage of the wildcard "*" to specify all resources within the Istio API group.
This role is generated by Istio installation by default.

### ClusterRole istiod-clusterrole-istio-system

> This ClusterRole allows full management of all Istio resources. This is done with the usage of the wildcard "*" to specify all resources within the Istio API group.

Reviewer: ```suggestion
This ClusterRole allows full management of all Istio resources. This is achieved by using the wildcard "*" to specify all resources within the Istio API group.
```

This role is generated by Istio installation by default.

> Additionally, this role allows read access to all resources from:

Reviewer: ```suggestion
Additionally, this role allows read access to all resources from the following groups:
```

- gateway.networking.k8s.io
- networking.k8s.io
- rbac.istio.io
This is required for Istiod to correctly maintain the state of the Istio service mesh.
Lastly, this role allows management over ingresses/status resources from networking.k8s.io API group.

> This is required by Istiod to manage the status of ingress resources.

Reviewer: ```suggestion
This is required for Istiod to correctly maintain the state of the Istio service mesh.
Lastly, this role allows management over ingresses and status resources from the `networking.k8s.io` API group. Istiod requires this permission to manage the status of ingress resources.
```
